File: Sqrt.java

public class Sqrt {
    public static int mySqrt(int x) {
        if (x == 0) {
            return 0;
        }

        int left = 1;
        int right = x;

        while (left <= right) {
            int mid = (left + right) / 2;
            int sqr = mid * mid;

            if (sqr == x) {
                return mid;
            } else if (sqr < x) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return right;
    }

    public static void testMySqrt() {
        
        int[][] testCases = {
            {0, 0},
            {4, 2},
            {8, 2},
            {9, 3},
            {16, 4},
            {25, 5},
            {30, 5},
            {36, 6}
        };

        for (int i = 0; i < testCases.length; i++) {
            int[] testCase = testCases[i];
            int input = testCase[0];
            int expected = testCase[1];
            int result = mySqrt(input);

            if (result == expected) {
                System.out.println("Test case " + (i + 1) + " passed!");
            } else {
                System.out.println("Test case " + (i + 1) + " failed. Expected " + expected + ", but got " + result + ".");
            }
        }
    }

    public static void main(String[] args) {
        testMySqrt();
    }
}

File: sqrt.py
def mySqrt(x):
    if x == 0:
        return 0

    left = 1
    right = x

    while left <= right:
        mid = (left + right) // 2
        sqr = mid * mid

        if sqr == x:
            return mid
        elif sqr < x:
            left = mid + 1
        else:
            right = mid - 1

    return right

def testMySqrt():
    
    testCases = [
        {"input": 0, "expected": 0},
        {"input": 4, "expected": 2},
        {"input": 8, "expected": 2},
        {"input": 9, "expected": 3},
        {"input": 16, "expected": 4},
        {"input": 25, "expected": 5},
        {"input": 30, "expected": 5},
        {"input": 36, "expected": 6},
    ]

    for i in range(len(testCases)):
        testCase = testCases[i]
        inputVal = testCase["input"]
        expected = testCase["expected"]
        result = mySqrt(inputVal)

        if result == expected:
            print(f"Test case {i + 1} passed!")
        else:
            print(f"Test case {i + 1} failed. Expected {expected}, but got {result}.")

testMySqrt()